2nd Semester/Homework 5/Routers/Routers/Graph.cs

```diff
@@ -0,0 +1,175 @@
+﻿namespace Routers;
+
+using Routers.Exceptions;
+
+/// <summary>
+/// Class that represents a graph.
+/// </summary>
+public class Graph
+{
+    private List<Edge> edges = new ();
+    private Dictionary<int, Node> nodes = new ();
+
+    /// <summary>
+    /// Gets dictionary that contains nodes of the graph with the keys being their index numbers.
+    /// </summary>
+    public IReadOnlyDictionary<int, Node> Nodes { get; private set; } = new Dictionary<int, Node>();
+
+    /// <summary>
+    /// Gets list containing edges of the graph.
+    /// </summary>
+    public IReadOnlyList<Edge> Edges { get; private set; } = new List<Edge>();
+
+    /// <summary>
+    /// Creates minimum spanning tree from a graph.
+    /// </summary>
+    /// <param name="path">Path to the input file.</param>
+    /// <returns>Minimum spanning tree.</returns>
+    /// <exception cref="GraphNotConnectedException">Throws if the graph was not connected.</exception>
+    /// <exception cref="ArgumentException">Throws if the input was not in correct form.</exception>
+    public static Graph MakeMinimalTree(string path)
+    {
+        var graph = new Graph();
+        graph.FillEdgesAndNodes(path);
+        graph.edges.Sort();
+        var tree = new Graph();
+        var newEdges = new List<Edge>();
+        foreach (var edge in graph.edges)
+        {
+            if (DisjointSet.Find(edge.Begin) != DisjointSet.Find(edge.End))
+            {
+                newEdges.Add(edge);
+                DisjointSet.Union(edge.Begin, edge.End);
+            }
+        }
+
+        tree.edges = newEdges;
+        tree.Edges = tree.edges;
+        tree.nodes = graph.nodes;
+        tree.Nodes = tree.nodes;
+        foreach (var node in tree.nodes)
+        {
+            node.Value.ConnectedEdges = new List<Edge>();
+            foreach (var edge in newEdges)
+            {
+                if (edge.Begin.Equals(node.Value) || edge.End.Equals(node.Value))
+                {
+                    node.Value.ConnectedEdges.Add(edge);
+                }
+            }
+        }
+
+        return tree;
+    }
+
+    private bool CheckForConnectivity()
+    {
+        var visitedNodes = new Dictionary<Node, bool>();
+        foreach (var node in nodes)
+        {
+            visitedNodes.Add(node.Value, false);
+        }
+
+        if (DepthFirstSearch(nodes[1], visitedNodes) == nodes.Count)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```

Comment:
```suggestion
        return DepthFirstSearch(nodes[1], visitedNodes) == nodes.Count;
```
Иначе это как if (x == 0) return 0 else if (x == 1) return 1 и т.д.